riscv/cachesim.h

```diff
@@ -60,18 +60,6 @@ class cache_sim_t
   void init(const std::string eviction_policy);
 };
 
-class fa_cache_sim_t : public cache_sim_t
-{
- public:
-  fa_cache_sim_t(size_t ways, size_t linesz, const char* name);
-  fa_cache_sim_t(size_t ways, size_t linesz, const char* name, const std::string eviction_policy);
-  uint64_t* check_tag(uint64_t addr);
-  uint64_t victimize(uint64_t addr);
- private:
-  static bool cmp(uint64_t a, uint64_t b);
-  std::map<uint64_t, uint64_t> tags;
-};
-
```

Comment:
If there isn't a compelling reason to make this change, I suggest backing it out.  My measurements suggest that for cache configuration `--ic=1:1024:32 --dc=1:1024:32`, with workload 403.gcc, there's a 2x perf hit if you use the naive linear-time algorithm.  (And the perf hit rises to 3x if I change `map` to `unordered_map`, which I guess I didn't know about at the time I wrote this.)

As an alternative, you could consider a more elaborate data structure that works well for both low-associativity and high-associativity caches, but that might be a bigger project.